src/hotspot/share/memory/metaspace.cpp

```diff
@@ -701,23 +704,49 @@ void Metaspace::global_initialize() {
   if (using_class_space() && !class_space_is_initialized()) {
     assert(!UseSharedSpaces, "CDS archive is not mapped at this point");
 
-    // case (b)
+    // case (b) (No CDS)
     ReservedSpace rs;
-
-    // If UseCompressedOops=1 and the java heap has been placed in coops-friendly
-    //  territory, i.e. its base is under 32G, then we attempt to place ccs
-    //  right above the java heap.
-    // Otherwise the lower 32G are still free. We try to place ccs at the lowest
-    // allowed mapping address.
-    address base = (UseCompressedOops && (uint64_t)CompressedOops::base() < OopEncodingHeapMax) ?
-                   CompressedOops::end() : (address)HeapBaseMinAddress;
-    base = align_up(base, Metaspace::reserve_alignment());
-
     const size_t size = align_up(CompressedClassSpaceSize, Metaspace::reserve_alignment());
-    if (base != NULL) {
-      if (CompressedKlassPointers::is_valid_base(base)) {
-        rs = ReservedSpace(size, Metaspace::reserve_alignment(),
-                           os::vm_page_size(), (char*)base);
+    address base = NULL;
+
+    // If CompressedClassSpaceBaseAddress is set, we attempt to force-map class space to
+    // the given address. This is a debug-only feature aiding tests. Due to the ASLR lottery
+    // this may fail, in which case the VM will exit after printing an appropiate message.
+    // Tests using this switch should cope with that.
+    if (!FLAG_IS_DEFAULT(CompressedClassSpaceBaseAddress) && CompressedClassSpaceBaseAddress != 0) {
```

Comment:
Don't you just have to test if CompressedClassSpaceBaseAddress != 0 , since that's the default if not set?

Reply:
Makes sense, I'll change that.